policy-controller/k8s/status/src/index.rs

```diff
@@ -0,0 +1,222 @@
+                    return false;
+                }
+                entry.insert(binding);
+            }
+        }
+        true
+    }
+
+    fn make_http_route_patch(
+        &self,
+        id: &ResourceId,
+        binding: &RouteBinding,
+    ) -> k8s::Patch<serde_json::Value> {
+        let accepting_servers: Vec<ResourceId> = self
+            .servers
+            .iter()
+            .filter(|server| binding.selects_server(server))
+            .cloned()
+            .collect();
+        let parent_statuses = binding
+            .parents
+            .iter()
+            .map(|parent| {
+                let ParentReference::Server(parent_reference_id) = parent;
+
+                // Is this parent in the list of parents which accept
+                // the route?
+                let accepted = accepting_servers
+                    .iter()
+                    .any(|accepting_parent| accepting_parent == parent_reference_id);
+                let condition = if accepted {
+                    k8s::Condition {
+                        last_transition_time: k8s::Time(Utc::now()),
+                        message: "".to_string(),
+                        observed_generation: None,
+                        reason: "Accepted".to_string(),
+                        status: "True".to_string(),
+                        type_: "Accepted".to_string(),
+                    }
+                } else {
+                    k8s::Condition {
+                        last_transition_time: k8s::Time(Utc::now()),
+                        message: "".to_string(),
+                        observed_generation: None,
+                        reason: "NoMatchingParent".to_string(),
+                        status: "False".to_string(),
+                        type_: "Accepted".to_string(),
+                    }
+                };
+                gateway::RouteParentStatus {
+                    parent_ref: gateway::ParentReference {
+                        group: Some(POLICY_API_GROUP.to_string()),
+                        kind: Some("Server".to_string()),
+                        namespace: Some(parent_reference_id.namespace.clone()),
+                        name: parent_reference_id.name.clone(),
+                        section_name: None,
+                        port: None,
+                    },
+                    controller_name: STATUS_CONTROLLER_NAME.to_string(),
+                    conditions: vec![condition],
+                }
+            })
+            .collect();
+        let status = gateway::HttpRouteStatus {
+            inner: gateway::RouteStatus {
+                parents: parent_statuses,
+            },
+        };
+        let patch_value = serde_json::json!({
+            "apiVersion": POLICY_API_VERSION,
+            "kind": "HTTPRoute",
+            "name": id.name,
+            "status": status,
+        });
+        k8s::Patch::Merge(patch_value)
+    }
+
+    fn apply_server_update(&self) {
```

Comment:
When we get a Server add or delete, we insert/remove the Server ID from `self.servers` but it doesn't look like we add or remove that Server from bindings in `self.http_routes`.  It seems like here we'd just build a patch from whatever the bindings in `self.http_routes` already were.

Reply:
The Servers in the index (`self.servers`) are different from those in each HTTPRoute's binding. We never want to remove a parent from an HTTPRoute's `RouteBinding.parents`; those represent the parent references which never change and are part of the HTTPRoute's spec.

When applying a Server update and therefore calling `make_http_route_patch` for each HTTPRoute, we collect all the Servers in the index that accept the HTTPRoute; we then look through the HTTPRoute's parents and check for any that have not accepted it.

Reply:
ah, I understand now, I had a misunderstanding about what the bindings in `self.http_routes` represented.  This makes sense but it's probably worth a comment on `http_routes` indicating that the bindings represent ALL parents refs in the HttpRoute, regardless of if they're valid or not.  We could additionally consider renaming the `RouteBinding` struct so that it doesn't suggest that these parents are "bound"... or even eliminate this type in favor of using `Vec<ParentReference>` directly.

Reply:
Okay I removed `RouteBinding` in favor of just using `Vic<ParentReference>` directly. I think this should clear up the understanding on why we we track an HTTPRoute's parent references and how those are used in this.